compiler/codegen.js

```diff
@@ -1767,6 +1767,14 @@ const disposeLeftover = wasm => {
 const generateExp = (scope, decl) => {
   const expression = decl.expression;
 
+  if (expression.type === 'Literal' && typeof expression.value === 'string') {
+    if (expression.value === 'use strict') {
+      if (scope.noStrict) return internalThrow(scope, 'SyntaxError', 'Illegal "use strict" directive', false);
```

Comment:
Curious, shouldn't the parser handle this?

Reply:
Some parsers have a .directive, and Babel even has some more complex things, but in general? Nothing stable.

Reply:
I meant that parsers should be able to detect this scenario and error for us, not directives generally.

Reply:
Ohhh, in my testing, no. But, maybe other parsers besides acorn do?

Reply:
Oh wait I misread, hmm strange.

Reply:
After some more testing, I'm coming to the same result. I'll go ahead and remove it.